mempool/src/core_mempool/mempool.rs

```diff
@@ -104,27 +126,32 @@ impl Mempool {
         txn: SignedTransaction,
         gas_amount: u64,
         ranking_score: u64,
-        db_sequence_number: u64,
+        crsn_or_seqno: AccountSequenceInfo,
         timeline_state: TimelineState,
         governance_role: GovernanceRole,
     ) -> MempoolStatus {
+        let db_sequence_number = crsn_or_seqno.min_seq();
         trace!(
             LogSchema::new(LogEntry::AddTxn)
                 .txns(TxnsLog::new_txn(txn.sender(), txn.sequence_number())),
             committed_seq_number = db_sequence_number
         );
         let cached_value = self.sequence_number_cache.get(&txn.sender());
-        let sequence_number =
-            cached_value.map_or(db_sequence_number, |value| max(*value, db_sequence_number));
+        let sequence_number = match crsn_or_seqno {
+            AccountSequenceInfo::CRSN { .. } => crsn_or_seqno,
+            AccountSequenceInfo::Sequential(_) => AccountSequenceInfo::Sequential(
+                cached_value.map_or(db_sequence_number, |value| max(*value, db_sequence_number)),
+            ),
+        };
         self.sequence_number_cache
-            .insert(txn.sender(), sequence_number);
+            .insert(txn.sender(), sequence_number.min_seq());
 
         // don't accept old transactions (e.g. seq is less than account's current seq_number)
-        if txn.sequence_number() < sequence_number {
+        if txn.sequence_number() < sequence_number.min_seq() {
```

Comment:
Below you say that txns won't be added to mempool unless they are in the open window. But here we only don't accept old transactions (left of min_nonce). Is that logic somewhere else?

Reply:
Good question. Yes, this check is done elsewhere in the prologue changes. When we call `CRSN::check` this will not allow the transaction to enter mempool unless it is within the window of the CRSN on-chain. I'll tag you in the place where that check is performed.

Reply:
Thanks. And I presume we do this (LHS, not LHS + window) check only here in case a new seqnonce has committed in the meantime?

Reply:
Also, the conditions for regular seqnonces was "For example, if the current sequence number for an account is 2 and local mempool contains transactions with sequence numbers 2, 3, 4, 7, 8, then only transactions 2, 3, and 4 will be broadcast.". What are the rules for CRSN txn broadcasting?